File: Jump Game II.js
function jump(nums) {
  let jumps = 0,
    currentEnd = 0,
    currentFarthest = 0;
  for (let i = 0; i < nums.length - 1; i++) {
    currentFarthest = Math.max(currentFarthest, i + nums[i]);
    if (i === currentEnd) {
      jumps++;
      currentEnd = currentFarthest;
    }
  }
  return jumps;
}

// Example usage:
const A = [2, 3, 1, 1, 4];
console.log(jump(A)); // Output: 2

File: Jump Game II.py
def jump(nums):
    jumps, current_end, current_farthest = 0, 0, 0
    for i in range(len(nums) - 1):
        current_farthest = max(current_farthest, i + nums[i])
        if i == current_end:
            jumps += 1
            current_end = current_farthest
    return jumps


# Example usage:
A = [2, 3, 1, 1, 4]
print(jump(A))  # Output: 2
